source/extensions/network/dns_resolver/cares/dns_impl.cc

```diff
@@ -356,6 +357,27 @@ void DnsResolverImpl::AddrInfoPendingResolution::startResolution() {
 }
 
 void DnsResolverImpl::AddrInfoPendingResolution::startResolutionImpl(int family) {
+  if (filter_unroutable_families_) {
+    switch (family) {
+    case AF_INET:
+      if (!available_interfaces_.v4_available_) {
```

Comment:
will `available_interfaces_.v4_available_` be `true` or `false` when the list of v4 interfaces contains `lo`?

Reply:
Good callout; we probably don't want to count loopback.

Reply:
Relatedly, in general you want to check v4/v6 availability on the interface you're going to use, any other interfaces are irrelevant

Reply:
added context below https://github.com/envoyproxy/envoy/pull/19287#discussion_r773340074